projects/akita-ng-fire/src/lib/collection/collection.service.ts

```diff
@@ -586,55 +590,59 @@ export class CollectionService<
     ids?: string[],
     options?: Partial<PathParams>
   ): Observable<EntityType[]>;
-  // tslint:disable-next-line: unified-signatures
   public valueChanges(
-    query?: QueryFn,
+    // tslint:disable-next-line: unified-signatures
+    queryConstraints?: QueryConstraint[],
     options?: Partial<PathParams>
   ): Observable<EntityType[]>;
   public valueChanges(
     id: string,
     options?: Partial<PathParams>
   ): Observable<EntityType>;
   public valueChanges(
-    idOrQuery?: string | string[] | QueryFn | Partial<PathParams>,
+    idOrQuery?: string | string[] | QueryConstraint[] | Partial<PathParams>,
     options: Partial<PathParams> = {}
   ): Observable<EntityType | EntityType[]> {
     const path = this.getPath(options);
     // If path targets a collection ( odd number of segments after the split )
     if (typeof idOrQuery === 'string') {
       const key = `${path}/${idOrQuery}`;
-      const query = () => this.db.doc<EntityType>(key).valueChanges();
-      return this.fromMemo(key, query).pipe(
-        map((doc) => this.formatFromFirestore(doc))
+      const docRef = doc(this.db, key) as DocumentReference<EntityType>;
+      const valueChangeQuery = () => docData<EntityType>(docRef);
+      return this.fromMemo(key, valueChangeQuery).pipe(
+        map(entity => this.formatFromFirestore(entity))
       );
     }
-    let docs$: Observable<EntityType[]>;
-    if (Array.isArray(idOrQuery)) {
-      if (!idOrQuery.length) return of([]);
+    let entities$: Observable<EntityType[]>;
+    if (Array.isArray(idOrQuery) && !idOrQuery.length) {
+      return of([]);
+    }
+    if (Array.isArray(idOrQuery) && typeof idOrQuery[0] === 'string') {
```

Comment:
It might be a good idea to create a guard function for that. Something like : 
```typescript
function isQueryConstraint(data: any): data is QueryConstract[] {
  return Array.isArray(data) && data.every(item => item instanceof QueryConstraint);
}
```